SU2_CFD/src/numerics/flow/flow_sources.cpp

```diff
@@ -562,3 +562,67 @@ CNumerics::ResidualType<> CSourceWindGust::ComputeResidual(const CConfig* config
 
   return ResidualType<>(residual, jacobian, nullptr);
 }
+
+CSourceRadiation::CSourceRadiation(unsigned short val_nDim, unsigned short val_nVar, const CConfig *config) :
+                  CSourceBase_Flow(val_nDim, val_nVar, config) {
+
+  implicit = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
+}
+
+CNumerics::ResidualType<> CSourceRadiation::ComputeResidual(const CConfig *config) {
+
+  unsigned short iDim;
+
+  /*--- Zero the continuity contribution. ---*/
+
+  residual[0] = 0.0;
+
+  /*--- Zero the momentum contribution. ---*/
+
+  for (iDim = 0; iDim < nDim; iDim++)
+    residual[iDim+1] = 0.0;
+
+  /*--- Set the energy contribution ---*/
+
+  residual[nDim+1] = -RadVar_Source[0]*Volume;
+
+  /*--- Set the energy contribution to the Jacobian. ---*/
+
+  if (implicit) {
+
+    /*--- Jacobian is set to zero on initialization. ---*/
+
+    jacobian[3][3] = -RadVar_Source[1]*Volume;
```

Comment:
@rsanfer why is the "3" hardcoded? Should this be nDim+1 to match the residual?

Reply:
Hi @pcarruscag ,
yes indeed, I missed this. I just updated it and will push the changes immediately. Thanks for catching it!